src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/grammar/java14/InputJava14LocalRecordAnnotation.java

```diff
@@ -0,0 +1,24 @@
+//non-compiled with javac: Compilable with Java14
+package com.puppycrawl.tools.checkstyle.grammar.java14;
+
+class InputJava14LocalRecordAnnotation {
+
+    public void doThing() {
+        // No error unless an annotation is present
+        @ProblemCauser
+        record Test(String hello, int world) {}
```

Comment:
please add few test cases with visibility modifers like "private/public" to see how they are located in AST under same `|--MODIFIERS`

Reply:
Local records cannot have visibility modifiers:
```
➜  src /usr/lib/jvm/java-14-openjdk/bin/javac --enable-preview --source 14 InputJava14LocalRecordAnnotation.java
InputJava14LocalRecordAnnotation.java:15: error: modifier public not allowed here
        public record Test2(String hello, int world) {}
               ^
Note: InputJava14LocalRecordAnnotation.java uses preview language features.
Note: Recompile with -Xlint:preview for details.
1 error
➜  src /usr/lib/jvm/java-14-openjdk/bin/javac --enable-preview --source 14 InputJava14LocalRecordAnnotation.java
InputJava14LocalRecordAnnotation.java:15: error: modifier private not allowed here
        private record Test2(String hello, int world) {}
                ^
Note: InputJava14LocalRecordAnnotation.java uses preview language features.
Note: Recompile with -Xlint:preview for details.
1 error
➜  src /usr/lib/jvm/java-14-openjdk/bin/javac --enable-preview --source 14 InputJava14LocalRecordAnnotation.java
InputJava14LocalRecordAnnotation.java:15: error: modifier protected not allowed here
        protected record Test2(String hello, int world) {}
                  ^
Note: InputJava14LocalRecordAnnotation.java uses preview language features.
Note: Recompile with -Xlint:preview for details.
1 error

```

Reply:
Please pay attention to link to source that I provided, I meant records as level of class fields, not inside methods.
If we have such already please share link

Reply:
>If we have such already please share link

I'm sure we do in a check update UT somewhere, but it's probably faster to just add some. 

Done.